client/human/HumanClientFSM.h

```diff
@@ -291,15 +296,25 @@ struct WaitingForGameStart : boost::statechart::state<WaitingForGameStart, Playi
     typedef boost::statechart::state<WaitingForGameStart, PlayingGame> Base;
 
     typedef boost::mpl::list<
+        boost::statechart::custom_reaction<DoneLoading>,
         boost::statechart::custom_reaction<GameStart>
     > reactions;
 
     WaitingForGameStart(my_context ctx);
     ~WaitingForGameStart();
 
+    boost::statechart::result react(const DoneLoading& msg);
     boost::statechart::result react(const GameStart& msg);
 
     CLIENT_ACCESSOR
+
+    //! Load data in an asyc thread.
+    void ProcessData(const Message& message);
+
+    //! Update UI in the GUI thread.
+    void ProcessUI(bool is_new_game, const SaveGameUIData& ui_data);
+
+    volatile bool is_loading = false;  //!< true from GameStart to DoneLoading
```

Comment:
Are you sure you know what you're doing with `volatile`?

http://cxx.isvolatileusefulwiththreads.com

Reply:
Atomic access is not needed here, so volatile is enough. Do you prefer `std::atomic`?

Reply:
If your answer is 'yes', that's fine.